### PR TITLE
Analyzer relax upper bound

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Review by mnonnenmacher — DISMISSED** [review]

**Review by fviernau — APPROVED** [review]

**Comment by sschuberth** [comment]
Doesn't seem to work, Azure still uses PHP 8.0 :-( https://dev.azure.com/oss-review-toolkit/ort/_build/results?buildId=2551&view=logs&j=a3985077-d6e4-50cc-d2e4-6de3ec48d9a5&t=00724c25-30b2-5aa4-698f-0646507c6655&l=1948

File: .azure-pipelines/LinuxAnalyzerTest.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ jobs:
     vmImage: ubuntu-18.04
   variables:
     FLUTTER_HOME: /opt/flutter
+    phpVersion: 7.2
```
Comment (mnonnenmacher) [inline .azure-pipelines/LinuxAnalyzerTest.yml:7]: According to these docs you also have to run `update-alternatives` several times:
https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/php?view=azure-devops#use-a-specific-php-version

**Comment by sschuberth** [comment]
While this PR doesn't fully fix the AppVeyor problem, I believe we should still merge it, and maybe again make the Azure Windows build mandatory instead of the AppVeyor one for the time being.

**Review by mnonnenmacher — APPROVED** [review]

**Comment by sschuberth** [comment]
Merging despite the AppVeyor failure, which will be addressed separately.

**Pull request merged**